sdk/core/Azure.Core/src/IJsonSerializable.cs

```diff
@@ -28,5 +28,19 @@ public interface IJsonSerializable
         /// <param name="options"></param>
         /// <returns></returns>
         bool TryDeserialize(Stream stream, out long bytesConsumed, SerializableOptions? options = default);
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="options"></param>
+        void Serialize(Stream stream, SerializableOptions? options = default);
```

Comment:
Yay!   

Personally, I'd prefer that these supersede the Try methods rather than exist side-by-side.   I don't know of a scenario where I'd want to use the Try approach.   We can always add that in if we see feedback asking for it.

Reply:
I think we need Try methods. I am not sure why we are adding non-try. The reason for Try methods is that when you deserialize JSON, you often don't know what's the next value to deserialize. For example, let's say you have an array of Animals and you dont know how many Animals are there. It's so much easier to keep trying to deserialize Animals till you cannot do this anymore and then just consume the "]" symbol.

Reply:
Resolving for now - but we will discuss this issue.